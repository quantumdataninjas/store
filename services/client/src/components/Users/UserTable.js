import React, { Component } from 'react'
import PropTypes from 'prop-types'
import {
  Grid,
  Table,
  TableHeaderRow
} from '@devexpress/dx-react-grid-material-ui'
import { Paper, Typography, withStyles } from '@material-ui/core'


const styles = theme => ({
  container: {
    // backgroundColor: "#353535",
    display: 'flex',
    flexWrap: 'wrap',
    padding: theme.spacing(3),
    justifyContent: 'center',
  },
  titlePaper: {
    backgroundColor: "#4b4b4b",
    display: 'flex',
    flexWrap: 'wrap',
    padding: theme.spacing(2),
    justifyContent: 'center',
    // fontSize: 40,
  },
  title: {
    fontSize: 40,
  },
  textField: {
    marginLeft: theme.spacing(1),
    marginRight: theme.spacing(1),
    width: 200,
  },
  dense: {
    marginTop: 19,
  },
  button: {
    background: 'linear-gradient(45deg, #FE6B8B 30%, #FF8E53 90%)',
    borderRadius: 3,
    border: 0,
    color: 'white',
    height: 50,
    padding: '0 30px',
    boxShadow: '0 3px 5px 2px rgba(255, 105, 135, .3)',
  },
})

class UserTable extends Component {

  static propTypes = {
    // users: PropTypes.array.isRequired,
    users: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number.isRequired,
      simple_user_id: PropTypes.number.isRequired,
      username: PropTypes.string.isRequired,
      email: PropTypes.string.isRequired,
      subscribed: PropTypes.bool.isRequired,
      terms_and_conditions: PropTypes.bool.isRequired,
      verified: PropTypes.bool.isRequired,
      firstname: PropTypes.string.isRequired,
      middlename: PropTypes.string,
      lastname: PropTypes.string.isRequired,
      // address1: PropTypes.string.isRequired,
      // address2: PropTypes.string,
      // city: PropTypes.string.isRequired,
      // state: PropTypes.string.isRequired,
      // zipcode: PropTypes.string.isRequired,
      // country: PropTypes.string.isRequired,
      phone: PropTypes.string,
      birthday: PropTypes.instanceOf(Date).isRequired,
      online: PropTypes.bool.isRequired,
      last_signin: PropTypes.instanceOf(Date).isRequired,
      last_signout: PropTypes.instanceOf(Date).isRequired,
      created_at: PropTypes.instanceOf(Date).isRequired
    })).isRequired
  }

  render() {
    let { classes, users } = this.props
    users.forEach(user => {
      Object.keys(user).forEach(key => {
        if(typeof(user[key]) === "boolean") {
          user[key] = user[key].toString()
        } else if(user[key] instanceof Date) {
          user[key] = user[key].toLocaleDateString()
        } else if(typeof(user[key]) === "undefined") {
          user[key] = "undefined"
        }
      })
    })
    return (
      <Paper className={classes.container}>
        <Grid
          rows={users}
          columns={[
            { name: 'id', title: 'ID' },
            { name: 'simple_user_id', title: 'Simple User ID' },
            { name: 'username', title: 'Username' },
            { name: 'email', title: 'Email' },
            { name: 'subscribed', title: 'Subscribed' },
            { name: 'terms_and_conditions', title: 'Terms And Conditions' },
            { name: 'verified', title: 'Verified' },
            { name: 'firstname', title: 'First Name' },
            { name: 'middlename', title: 'Middle Name' },
            { name: 'lastname', title: 'Last Name' },
            // { name: 'address1', title: 'Address1' },
            // { name: 'address2', title: 'Address2' },
            // { name: 'city', title: 'City' },
            // { name: 'state', title: 'State' },
            // { name: 'zipcode', title: 'Zipcode' },
            // { name: 'country', title: 'Country' },
            { name: 'phone', title: 'Phone' },
            { name: 'birthday', title: 'Birthday' },
            { name: 'online', title: 'Online' },
            { name: 'last_signin', title: 'Last Signin' },
            { name: 'last_signout', title: 'Last Signout' },
            { name: 'created_at', title: 'Created At' },
            { name: 'deleted', title: 'Deleted' },
            { name: 'deleted_at', title: 'Deleted At' },
          ]}>
          <Paper
            className={classes.titlePaper}
          >
            {/* <h1 className="title is-1" style={{color: 'white'}}>
              Users
            </h1> */}
            <Typography variant="h1" component="h2" className={classes.title}>
              Users
            </Typography>
          </Paper>
          {/* <Table style={{borderRight: "white"}} /> */}
          <Table />
          <TableHeaderRow />
        </Grid>
      </Paper>
    )
  }
}

export default withStyles(styles)(UserTable)
