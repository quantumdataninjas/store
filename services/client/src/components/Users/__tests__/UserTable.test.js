import React from 'react'
import { mount, shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import UserTable from '../UserTable'


const users = [
    {
        "id": "1",
        "email": "user@test.org",
        "subscribed": "true",
        "terms_and_conditions": "true",
        "verified": "false",
        "firstname": "first",
        "middlename": "",
        "lastname": "last",
        "address1": "1523 John St.",
        "address2": "",
        "city": "Fort Lee",
        "state": "NJ",
        "zipcode": "07024",
        "country": "United States",
        "phone": "",
        "birthday": "1/1/1990",
        "online": "true",
        "last_signin": "8/26/2019",
        "last_signout": "8/26/2019",
        "created_at": "8/26/2019"
    },
    {
        "id": "2",
        "email": "user2@test.org",
        "subscribed": "true",
        "terms_and_conditions": "true",
        "verified": "false",
        "firstname": "first",
        "middlename": "",
        "lastname": "last",
        "address1": "1523 John St.",
        "address2": "",
        "city": "Fort Lee",
        "state": "NJ",
        "zipcode": "07024",
        "country": "United States",
        "phone": "",
        "birthday": "1/1/1990",
        "online": "true",
        "last_signin": "8/26/2019",
        "last_signout": "8/26/2019",
        "created_at": "8/26/2019"
    },
]

const cols = [
    { header: "ID", name: "id" },
    { header: "Email", name: "email" },
    { header: "Subscribed", name: "subscribed" },
    { header: "Terms And Conditions", name: "terms_and_conditions" },
    { header: "Verified", name: "verified" },
    { header: "First Name", name: "firstname" },
    { header: "Middle Name", name: "middlename" },
    { header: "Last Name", name: "lastname" },
    { header: "Address1", name: "address1" },
    { header: "Address2", name: "address2" },
    { header: "City", name: "city" },
    { header: "State", name: "state" },
    { header: "Zipcode", name: "zipcode" },
    { header: "Country", name: "country" },
    { header: "Phone", name: "phone" },
    { header: "Birthday", name: "birthday" },
    { header: "Online", name: "online" },
    { header: "Last Signin", name: "last_signin" },
    { header: "Last Signout", name: "last_signout" },
    { header: "Created At", name: "created_at" },
]

describe('UserTable Component', () => {

    it('renders empty message as table cell if there is no data', () => {
        const wrapper = mount(<UserTable users={[]}/>)
        const table = wrapper.find('table');
        expect(table).toHaveLength(1);
        const thead = table.find('thead');
        expect(thead).toHaveLength(1);
        const headers = thead.find('th');
        expect(headers).toHaveLength(cols.length);
        headers.forEach((th, idx) => {
            expect(th.text()).toEqual(cols[idx].header);
        });
        const tbody = table.find('tbody');
        expect(tbody).toHaveLength(1);
        const row = tbody.find('tr');
        expect(row).toHaveLength(1);
        const cell = row.find('td');
        expect(cell).toHaveLength(1);
        expect(cell.prop('colSpan')).toEqual(cols.length);
        expect(cell.text()).toEqual('No data');
    })

    it('renders properly', () => {
        const wrapper = mount(<UserTable users={users}/>)
        const table = wrapper.find('table')
        expect(table).toHaveLength(1)
        const thead = table.find('thead')
        expect(thead).toHaveLength(1)
        const headers = thead.find('th')
        expect(headers).toHaveLength(cols.length)
        headers.forEach((th, i) => {
            expect(th.text()).toEqual(cols[i].header)
        })
        const tbody = table.find('tbody')
        expect(tbody).toHaveLength(1)
        const rows = tbody.find('tr')
        expect(rows).toHaveLength(users.length)
        rows.forEach((tr, i) => {
            const cells = tr.find('td')
            expect(cells).toHaveLength(cols.length)
            cells.forEach((cell, j) => {
                expect(cell.text()).toEqual(users[i][cols[j].name])
            })
        })
    })

    it('renders a snapshot', () => {
        const tree = renderer.create(<UserTable users={users} />)
        expect(tree).toMatchSnapshot()
    })
})