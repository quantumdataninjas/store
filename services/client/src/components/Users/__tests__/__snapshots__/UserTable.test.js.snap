// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTable Component renders a snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <h1
    className="title is-1"
    style={
      Object {
        "color": "white",
      }
    }
  >
    User Table
  </h1>
  <div
    className="RootBase-root-28"
  >
    <div
      className="TableContainer-root-29"
    >
      <div>
        <table
          className="MuiTable-root Table-table-30"
          style={
            Object {
              "minWidth": "2400px",
            }
          }
        >
          <colgroup>
            <col />
            <col />
            <col />
            <col />
            <col />
            <col />
            <col />
            <col />
            <col />
            <col />
            <col />
            <col />
            <col />
            <col />
            <col />
            <col />
            <col />
            <col />
            <col />
            <col />
          </colgroup>
          <thead
            className="MuiTableHead-root"
          >
            <tr
              className="MuiTableRow-root MuiTableRow-head"
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      ID
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Email
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Subscribed
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Terms And Conditions
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Verified
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      First Name
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Middle Name
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Last Name
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Address1
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Address2
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      City
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      State
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Zipcode
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Country
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Phone
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Birthday
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Online
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Last Signin
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Last Signout
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Created At
                    </span>
                  </div>
                </div>
              </th>
            </tr>
          </thead>
          <tbody
            className="MuiTableBody-root"
          >
            <tr
              className="MuiTableRow-root"
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                1
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                user@test.org
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                true
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                true
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                false
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                first
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                last
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                1523 John St.
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                Fort Lee
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                NJ
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                07024
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                United States
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                1/1/1990
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                true
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                8/26/2019
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                8/26/2019
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                8/26/2019
              </td>
            </tr>
            <tr
              className="MuiTableRow-root"
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                2
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                user2@test.org
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                true
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                true
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                false
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                first
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                last
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                1523 John St.
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                Fort Lee
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                NJ
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                07024
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                United States
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                1/1/1990
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                true
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                8/26/2019
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                8/26/2019
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                8/26/2019
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
